src/request.m

```diff
@@ -295,8 +295,11 @@ - (void) _updateWindow: (NSInteger) delta withBlock: (void (^)()) block {
       window_out_queue_ = [NSMutableArray arrayWithCapacity: 16];
 
     // Retry on positive window_out
+    __weak __typeof(self) weakSelf = self;
```

Comment:
@cristiankocza `_windowUpdate` is a way to unqueue accumulated data from request. Using `weak` here will probably cause some data to be not delivered...